dot/peerset/peerset_test.go

```diff
@@ -125,58 +120,60 @@ func TestReAllocAfterBanned(t *testing.T) {
 	}
 
 	// We ban a node by setting its reputation under the threshold.
+	processor.EXPECT().Process(Message{Status: Drop, setID: 0, PeerID: peer1})
 	rep := newReputationChange(BannedThresholdValue-1, "")
+
 	// we need one for the message to be processed.
+	processor.EXPECT().Process(Message{Status: Reject, setID: 0, PeerID: peer1})
 	handler.ReportPeer(rep, peer1)
 	time.Sleep(time.Millisecond * 100)
 
-	checkMessageStatus(t, <-ps.resultMsgCh, Drop)
-
 	// Check that an incoming connection from that node gets refused.
-
+	processor.EXPECT().Process(Message{Status: Connect, setID: 0, PeerID: peer1})
 	handler.Incoming(0, peer1)
```

Comment:
put assert/require statements to show impact of handler.ReportPeer and handler.Incoming